# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/TextDirectiveCreator.h
# Commit: 29735d438f26
# Full Hash: 29735d438f2634e2c6c1d8e80857bfb144d36483
# Author: Jan-Niklas Jaeschke <jjaschke@mozilla.com>
# Date: 2025-05-27 03:54:23
# Regressor Bug: 1957557
# File Overlap Count: 1
# Description:
#   Bug 1957557, part 7 - Create Text Fragments: Compute word boundary distances for context terms. r=dom-core,farre
#   
#   The distances are always in search direction:
#   For prefix and end terms, the positions are
#   relative to the end of the term and use the word
# ==============================================================================

diff -r 3c7adffd2377 -r 29735d438f26 dom/base/TextDirectiveCreator.h
--- a/dom/base/TextDirectiveCreator.h	Mon May 26 08:18:28 2025 +0000
+++ b/dom/base/TextDirectiveCreator.h	Mon May 26 08:18:28 2025 +0000
@@ -96,14 +96,32 @@
    */
   Result<Ok, ErrorResult> CollectSuffixContextTerm();
 
+  /**
+   * @brief Collect the word begin / word end distances for the context terms.
+   *
+   * For start (for range-based matching) and suffix terms, the search direction
+   * is left-to-right. Therefore, the distances are based off the beginning of
+   * the context terms and use the word end boundary.
+   *
+   * For prefix and end (for range-based matching), the search direction is
+   * right-to-left. Therefore, the distances are based off the end of the
+   * context terms and use the word start boundary.
+   *
+   * The distances are always sorted, so that the first entry points to the
+   * nearest word boundary in search direction.
+   */
+  virtual void CollectContextTermWordBoundaryDistances() = 0;
+
   nsString mPrefixContent;
   nsString mPrefixFoldCaseContent;
+  nsTArray<uint32_t> mPrefixWordBeginDistances;
 
   nsString mStartContent;
   nsString mStartFoldCaseContent;
 
   nsString mSuffixContent;
   nsString mSuffixFoldCaseContent;
+  nsTArray<uint32_t> mSuffixWordEndDistances;
 
   Document& mDocument;
   RefPtr<AbstractRange> mRange;
@@ -128,8 +146,13 @@
 
   Result<Ok, ErrorResult> CollectContextTerms() override;
 
+  void CollectContextTermWordBoundaryDistances() override;
+
   nsString mEndContent;
   nsString mEndFoldCaseContent;
+
+  nsTArray<uint32_t> mStartWordEndDistances;
+  nsTArray<uint32_t> mEndWordBeginDistances;
 };
 
 /**
@@ -142,6 +165,7 @@
 
   Result<Ok, ErrorResult> CollectContextTerms() override;
 
+  void CollectContextTermWordBoundaryDistances() override;
 };
 }  // namespace mozilla::dom
 #endif